test/autorest/complexgroup/complexgroup_test.go

```diff
@@ -67,3 +75,85 @@ func TestPutValid(t *testing.T) {
 	}
 	deepEqualOrFatal(t, result, expected)
 }
+
+// TODO check this
+func TestGetInvalid(t *testing.T) {
```

Comment:
Wondering what to do here, since the test specifically sends a string instead of an integer and obviously that cannot be unmarshalled into the Basic model

Reply:
So unmarshalling should fail and an error is returned right?  If that's the case then ensure that the response object is nil and the error is non-nil.  At some point we'll need to ensure that the returned errors are actionable.

Reply:
Ah perfect, I'll just add the check for response object

Reply:
Done